logging/net_appender.go

```diff
@@ -84,6 +85,34 @@ type NetAppender struct {
 	loggerWithoutNet Logger
 }
 
+func (w *remoteLogWriterGRPC) setConn(ctx context.Context, logger Logger, conn rpc.ClientConn, sharedConn bool) {
+	w.clientMutex.Lock()
+	defer w.clientMutex.Unlock()
+	if w.rpcClient != nil && !w.sharedConn {
+		oldClient := w.rpcClient
+		go func() {
+			err := oldClient.Close()
+			logger.CErrorf(ctx, "error closing oldClient: %s", err)
+		}()
```

Comment:
is there a fear that closing may hang?

Reply:
no reason to think Close is slow -- should I remove the coroutine and do this inline?

Reply:
ya just do it inline unless we are running into issues

Reply:
done in 4019838